src/twilio_status_callback_handler.ts

```diff
@@ -0,0 +1,51 @@
+import { logTwilioStatusToDb } from './db_api_util';
+import logger from './logger';
+import { addSlackMessageReaction } from './slack_api_util';
+import { Request } from './types';
+
+// Twilio statuses that represent the final state of a message. The value
+// in this map is the emoji reaction to add to the Slack message.
+const FINAL_STATUSES: { [status: string]: string } = {
+  delivered: 'white_check_mark',
+  undelivered: 'x',
+  failed: 'x',
+};
+
+export async function handleTwilioStatusCallback(req: Request): Promise<void> {
+  const { MessageStatus: messageStatus, MessageSid: messageSid } = req.body;
+
+  logger.info(
+    `HANDLING TWILIO CALLBACK: message sid ${messageSid} has status ${messageStatus}`
+  );
+
+  // Twilio delivers callbacks for lots of intermediate states (queued, sent
+  // to carrier, etc.). We just care about the final status of the message,
+  // so we don't update Postgres for those intermediate statuses.
+  if (!(messageStatus in FINAL_STATUSES)) {
+    logger.info(`Twilio status is non-final; not processing`);
+    return;
+  }
+
+  // Update the Postgres DB with the message status and current timestamp
+  const slackMessageInfo = await logTwilioStatusToDb(messageSid, messageStatus);
```

Comment:
Out of curiosity, why doesn't this need a catch? I see the function itself has a try/finally within it like other DBApiUtil functions, but I think we have try/catches around calls to other DBApiUtil functions elsewhere.

Reply:
We only need a try/catch if there's something we want to do when there's an error other than just return a 500. In this case, there's nothing to do -- if we can't query the database, we can't get the slack message info, so we can't add a reaction. So we might as well just let the error bubble up and return a 500 -- try/catch is just for when there's some useful fallback behavior or graceful failure mode.

Reply:
What about catching so we can log to Sentry?